src/data/model/mutable/MutableObject.ts

```diff
@@ -251,7 +251,8 @@ abstract class MutableObject extends HashedObject {
                 const checkpoint = await this.getStore().loadLastCheckpoint(this.getLastHash());
 
                 if (checkpoint !== undefined) {
-                    this.restoreCheckpoint(checkpoint);
+                    console.log('Restoring checkpoint for ' + this.getClassName() + ' ' + this.getLastHash());
+                    await this.restoreCheckpoint(checkpoint);
```

Comment:
@sbazerque this is happening but the changes don't propagate to the UI. I think I need to consult with you about what should happen after this.